bodo/pandas/physical/reduce.h

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include <arrow/compute/expression.h>
+#include <arrow/scalar.h>
+#include <arrow/type_fwd.h>
+#include <memory>
+#include <stdexcept>
+#include <utility>
+#include "../libs/_bodo_to_arrow.h"
+#include "operator.h"
+
+#undef CHECK_ARROW
+#define CHECK_ARROW(expr, msg)                                                \
+    if (!(expr.ok())) {                                                       \
+        std::string err_msg = std::string("PhysicalReduce::ConsumeBatch: ") + \
+                              msg + " " + expr.ToString();                    \
+        throw std::runtime_error(err_msg);                                    \
+    }
+
+/**
+ * @brief Physical node for reductions like max.
+ *
+ */
+class PhysicalReduce : public PhysicalSource, public PhysicalSink {
+   public:
+    explicit PhysicalReduce(std::shared_ptr<bodo::Schema> in_table_schema,
+                            std::string function_name)
+        // Drop Index columns since not necessary in output
+        : out_schema(in_table_schema->Project(1)),
+          function_name(function_name),
+          scalar_cmp_name(getScalarCmpName(function_name)) {
+        // TODO: support reductions that out different output types from input
+        // types (e.g. upcast in sum)
```

Comment:
throw if not the same type?

Reply:
We are setting output type the same as input type here. Different types only possible in accumulative cases like sum that are not implemented.